File: TestSample/cpp/src/speakers/JaSpeaker.cpp
#include "JaSpeaker.h"

namespace helloworld::speakers {

JaSpeaker::JaSpeaker(std::string name)
    : Person(std::move(name), "ja") {}

std::string JaSpeaker::speak() const {
    return "こんにちは、世界！";
}

}

File: TestSample/php/src/Speakers/JaSpeaker.php
<?php
declare(strict_types=1);

namespace HelloWorld\Speakers;

use HelloWorld\Person;

final class JaSpeaker extends Person
{
    public function __construct(string $name)
    {
        parent::__construct($name, "ja");
    }

    public function speak(): string
    {
        return "こんにちは、世界！";
    }
}
